### ISSUE TITLE
Flux seems slightly confused with an implicit "latest" tag

### ISSUE THREAD

**Issue by dlespiau** [fluxcd/flux#929]
I have a deployment with the image name not containing any tag: https://github.com/dlespiau/flux-test/blob/02441c13d4d6be2c3132b23970cdc77f53c698d7/quay-exporter-deploy.yaml#L18

The UI isn't showing that the "latest" is deployed in that case:

![screenshot from 2018-02-05 12-14-40](https://user-images.githubusercontent.com/7986/35804088-2a77ca72-0a6e-11e8-8d9d-f4b0fe2f9866.png)

I would have expected "latest" to be shown as "current".

**Comment by squaremo** [comment]
Well .. `:latest` in that list _isn't_ deployed, at least not necessarily. You can have an image with no `latest` tag and still run an untagged image, if an image from that repo is already present on the host. It's a bit of a tangle.

**Comment by dlespiau** [comment]
Ah, I see. That makes sense.

I wonder if the `Image ID` field from `kubectl describe pod` could be used in that case as the true image source.

```
$ kubectl describe pod quay-exporter-8477cd5649-b6qnb
Name:		quay-exporter-8477cd5649-b6qnb
Namespace:	default
Node:		launcher-tests/192.168.99.100
Start Time:	Mon, 05 Feb 2018 13:09:09 +0000
Labels:		name=quay-exporter
		pod-template-hash=4033781205
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"default","name":"quay-exporter-8477cd5649","uid":"c09d13e2-0a75-11e8-8dd1-0800279...
Status:		Running
IP:		172.17.0.5
Created By:	ReplicaSet/quay-exporter-8477cd5649
Controlled By:	ReplicaSet/quay-exporter-8477cd5649
Containers:
  quay-exporter:
    Container ID:	docker://a128a1307cc9e7746d7287eaa9b18a06f4e851f03cb06a7c2ae90d33eba24b19
    Image:		quay.io/damien.lespiau/quay-exporter
    Image ID:		docker-pullable://quay.io/damien.lespiau/quay-exporter@sha256:a02a2f44f8c9f89ddff37faf5435d2979097d5baaa05ec342c01946b75d5a101
```

**Comment by kingdonb** [comment]
This might be resolved by #3440

**Comment by kingdonb** [comment]
It does appear to be fixed with #3440, I think this can be closed with the merge 🎉 

<img width="896" alt="Screen Shot 2021-03-10 at 5 16 06 PM" src="https://user-images.githubusercontent.com/3286998/110705713-c6929380-81c4-11eb-9139-ba65a5abea43.png">
